File: src/leetcode/283. Move Zeroes/move_zeroes283.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int k = 0;
        for(int i = 0; i < n; i++) {
            if(nums[i] != 0) {
                int tmp = nums[k];
                nums[k] = nums[i];
                nums[i] = tmp;
                k++;
            }
        }
    }
};

File: src/leetcode/solved-collection/283. Move Zeroes/move_zeroes283.go
func moveZeroes(nums []int)  {
    k := 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != 0 {
            nums[i], nums[k] = nums[k], nums[i]
            k++
        }
    }
}
